### PR TITLE
adding review process document

### PR DESCRIPTION
**Pull request by bardkw**
The purpose of this PR is to document our review process.

### CODE REVIEW

File: docs/review_process.rst [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+============
+
+Once the automatic checks are either resolved or addressed, the submitted PR will need to go through a human review. Reviewers should add comments to provide feedback and raise potential issues. Should the PR pass their review, the reviewer should then indicate that it has their approval.
+
+A review can still be requested before the checks are resolved, but the PR must be marked as a ``Draft``. Once the PR is in a mergable state, it will need to undergo a final review to ensure that there are no outstanding issues.
+
+CI/MyPy/Pytest
+==============
+
+* ``lint`` runs flake8 to quality check the code. You can run this check manually in your local repository with ``poetry run flake8``.
```
Comment (Purg) [inline docs/review_process.rst:51]:
```suggestion
* ``lint`` runs flake8 to quality check the code style. You can run this check manually in your local repository with ``poetry run flake8``.
```

File: docs/review_process.rst [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+* ``MyPy`` performs a type check. You can run this check manually in your local respository with ``poetry run mypy``.
```
Comment (Purg) [inline docs/review_process.rst:53]:
```suggestion
* ``MyPy`` performs static type analysis. You can run this check manually in your local repository with ``poetry run mypy``.
```

File: docs/review_process.rst [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+The purpose of this document is to define the process for reviewing and integrating branches into SMQTK. This encompasses all SMQTK repositories.
+
+See ``CONTRIBUTING.md`` for guidelines on contributing to SMQTK.
+
+See `release_process`_ for guidelines on the release process for SMQTK.
+
+.. _`release_process`: release_process.html
+
+.. contents:: The review process consisists of the following steps:
```
[the remark this answers is not in the document]
Comment (Purg) [reply docs/review_process.rst:12]:
```suggestion
.. contents:: The review process consists of the following steps:
```

File: docs/review_process.rst [hunk 4]
```diff
@@ -0,0 +1,69 @@
+SMQTK Review Process
+********************
+
+The purpose of this document is to define the process for reviewing and integrating branches into SMQTK. This encompasses all SMQTK repositories.
+
+See ``CONTRIBUTING.md`` for guidelines on contributing to SMQTK.
+
+See `release_process`_ for guidelines on the release process for SMQTK.
```
[the remark this answers is not in the document]
Comment (bardkw) [reply docs/review_process.rst:8]: No I was referring to the release process for new releases while this exclusively covers PR's

File: docs/review_process.rst (hunk 3 above)
[the remark this answers is not in the document]
Comment (bardkw) [reply docs/review_process.rst:12]: I think contents makes sense for larger files. Looking at this now, I think i want to adjust the depth so it looks more concise.

File: docs/review_process.rst (hunk 4 above)
[the remark this answers is not in the document]
Comment (Purg) [reply docs/review_process.rst]: Oops! Apparently I dislexia'd "release" for "review".

**Review by Purg — APPROVED** [review]

**Pull request merged**